### ISSUE TITLE
scaffold still resizes even when resizeToAvoidBottomInset is set to false on mobile Web platform

### ISSUE THREAD

**Issue by kean-fetchpet** [flutter/flutter#133433]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. run the code below 
2. run flutter build web --release
3. open the website on mobile.

I also hosted the code below and you can access via this link https://timezone-dot-dev-public-383006.ts.r.appspot.com/#/. 

I do not get this issue for all the mobile web browsers. I tried with iphone 13 safari browser and its fine (the keyboard obscures the text field which is intended). However, I tested with pixel 5 phone (Android Version 13) and got this issue . 

### Expected results

From the flutter documentation, when resizeToAvoidBottomInset i set to false, scaffold should not be resized and the text field should be obscured. 

### Actual results

Text field is pushed to above keyboard. 

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';


void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Timezone'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  TextEditingController controller = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      resizeToAvoidBottomInset: false,
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Padding(
        padding: const EdgeInsets.all(20.0),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.end,
          children: [
            Padding(
              padding: const EdgeInsets.symmetric(vertical: 20),
              child: TextField(
                  controller: controller,
                  cursorColor: Colors.pink,
                  style: const TextStyle(
                  color: Colors.pink
              )),
            ),
          ]
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

https://github.com/flutter/flutter/assets/116161269/ca8fb57f-d04f-45b3-a041-82c33a8968e0

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.3.1 22E261 darwin-arm64, locale en-AU)
    • Flutter version 3.10.6 on channel stable at /Users/keantan/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (7 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/keantan/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] Connected device (4 available)
    • iPhone SE (3rd generation) (mobile) • BEBC05D7-14CD-4B1E-96AD-150ACB421900 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • iPhone 14 Pro (mobile)              • BD90EE10-6741-462F-881B-E6869CBC29EB • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)                     • macos                                • darwin-arm64   • macOS 13.3.1 22E261 darwin-arm64
    • Chrome (web)                        • chrome                               • web-javascript • Google Chrome 116.0.5845.110

[✓] Network resources
    • All expected network resources are available.


```

</details>

**Comment by kevmoo** [comment]
I doubt I'd fix this. 😄  Should wait for our weekly triage to take a look.

**Comment by kean-fetchpet** [comment]
Yes please. Thanks.

**Comment by kean-fetchpet** [comment]
@kevmoo just wondering what is the turnaround time for your team to look into this? Would be great to know this as it will tell us if we need to create a separate design for android users. Thanks.

**Comment by kevmoo** [comment]
Are we sure this is web only?

**Comment by foo-fetchpet** [comment]
So we've performed the test on the following emulators and they both had no issues:
- Pixel 4
- Pixel 5

We've also tested on the following physical devices using wireless debug and they all had no issues:
- Pixel 5
- Pixel 7 Pro
- Samsung Galaxy S20
- Samasung Galaxy S10

We've also used:
"flutter run -d web-server --web-port=8080 --web-hostname=0.0.0.0 "
Then going to 192.168.X.X:8080 on each mobile's chrome browser and they all had the issue of the keyboard coming up, the textbox sitting on top of the keyboard, visible on screen, and the app bar "Flutter Timezone" moving off the top of the screen:
- Pixel 5
- Pixel 7 Pro
- Samsung Galaxy S20
- Samsung Galaxy S10

https://github.com/flutter/flutter/assets/116143436/cd940fbf-f616-4004-9e5e-3b0006bd9f22

Our guess is this is a mobile web issue.

**Comment by htoor3** [comment]
Thanks for the details @foo-fetchpet and helping narrow this down.  

So it appears to be only reproducible on physical Android devices + chrome? I'll try to repro this on a Pixel and investigate further.

**Comment by htoor3** [comment]
I've reproduced this via remote debugging an android device (Pixel 6, Android 13, Chrome)
- I've tried using the same trick from the `IOSTextEditingStrategy` to delay placement of the text field, but it doesn't change anything.  
- I've tested this with both `true` and `false` for `resizeToAvoidBottomInset` and there's a very subtle difference.  When `true`, the entire scaffold resizes and the headers remains static.  When `false` (when the bug is occurring), the field gets pushed up and the header slightly scrolls out of view.  
- I'm not sure this is a text field issue as much as it's a scaffold resize issue and a bug with the `resizeToAvoidBottomInset` property.  Adding a `framework` label to see if the framework team has any insights on this.

**Comment by darshankawar** [comment]
Using the code sample [provided](https://github.com/flutter/flutter/issues/133433#issue-1869111776), I was able to replicate this on latest versions (stable and master) running on mobile web but only on Android + chrome, wherein, we get to see below behavior with flag set to `false`:


https://github.com/flutter/flutter/assets/67046386/21c48d83-b981-4941-9e47-d9d15c1bd774


With flag set to `true`, it works as expected:


https://github.com/flutter/flutter/assets/67046386/d0b519c3-ec25-4ada-9137-7c5dbd47fbc9


The same behavior works as expected running on mobile browser iOS safari with flag set to false, so this seems to be specific to Android + chrome.


<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.19.5, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.19.5 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 300451adae (4 days ago), 2024-03-27 21:54:07 -0500
    • Engine revision e76c956498
    • Dart version 3.3.3
    • DevTools version 2.31.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.22.0-5.0.pre.55, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.22.0-5.0.pre.55 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ddf79b755d (6 hours ago), 2024-04-07 18:34:24 -0400
    • Engine revision c10e950c66
    • Dart version 3.5.0 (build 3.5.0-18.0.dev)
    • DevTools version 2.34.1
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by htoor3** [comment]
Thanks for re-triaging this @darshankawar, I'll go ahead and give this a priority since there's a discrepancy in behavior between Android and iOS.

**Comment by ReinisSprogis** [comment]
It has been a while. A fix would be welcome. Android + chrome must be affecting many users. For me also AlertDialog and BottomSheet is jumping up. Sometimes it does, sometimes it doesn't.

**Comment by Keetz** [comment]
Also experiencing issues on Android + Chrome.

I am testing with a TextField in a ModalBottomSheet.
`resizeToAvoidBottomInset: false` is set on the Scaffold.

As a workaround I do this: `kIsWeb ? 0 : MediaQuery.of(context).viewInsets.bottom`.
It only fixes the part where it pushes content even further up if I always add the viewInsets.bottom padding from MediaQuery.

It does not fix that it pushes the content behind the modal sheet up even though resizeToAvoidBottomInset is set to false. Notice the `Android + Chrome, with 0 bottom padding` image and how the buttons are pushed up.

No modal sheet opened yet:
![1000001848](https://github.com/flutter/flutter/assets/6707330/a005950e-91b9-48fd-9aa6-c31f76f057c6)

Android native:
![1000001845](https://github.com/flutter/flutter/assets/6707330/58164402-0b3c-473a-b2e8-54fb89ed703f)

Android + Chrome, with 0 bottom padding:
![1000001844](https://github.com/flutter/flutter/assets/6707330/e6033aa9-0937-432b-b557-34bf329ab035)

Android + Chrome, with MediaQuery.of(context).viewInsets.bottom padding:
![1000001846](https://github.com/flutter/flutter/assets/6707330/deba1328-2dee-44e9-8262-c9c147f89574)

The last one has mixed results, sometimes it pushes the sheet even further up and out of the screen:
![1000001847](https://github.com/flutter/flutter/assets/6707330/b8d2edf5-d077-4072-8e6f-81b015382f10)

**Comment by Keetz** [comment]
Tested my above example on iOS Safari and it works just as perfect as Android native, so definitely seems to be a Android + Chrome issue

**Comment by ReinisSprogis** [comment]
https://github.com/flutter/flutter/assets/69913791/2674c415-0fbd-42c6-885c-84d7546eb4c8

https://[playgroundwoltexample.web.app](https://playgroundwoltexample.web.app/)/
I experience the same in my app. Android + chrome

**Comment by lisovyk** [comment]
Still a problem!

**Comment by andreykuzovlevv** [comment]
still problem on iOS

**Comment by EbenezerGH** [comment]
Still a problem on my pixel 9 :(.  Seeing as this has been here since Aug 2023, i'm not sure i'll be able to get a solution before a launch this year.

**Comment by nvbien2000** [comment]
It's been a while, any update on this? I used `resizeToAvoidBottomInset: false` on parent widget's scaffold also, but it works well in iOS only.

**Comment by Mweo-Bruce** [comment]
Having the same issue as well on Android Pixel 9, resizeToAvoidBottomInset: false doesn't work on parent widget's scaffold

**Comment by zylthinking** [comment]
Same problem(s) found in flutter web :

When resizeToAvoidBottomInset: true:
The UI moves up when the soft keyboard appears. I can see the MaterialApp.scaffoldBackgroundColor under the keyboard.

When resizeToAvoidBottomInset: false:
The UI still moves up when the soft keyboard appears, but the bottom slice of the UI remains covered (it doesn't move up enough). The color under the keyboard is the background color of the moved-up UI.

On iOS with resizeToAvoidBottomInset: false:
When tapping the editor, the keyboard appears but the UI doesn't move up at all.
However, if I long-touch the editor (which gives focus without popping up the keyboard), then tapping the focused editor again causes the UI to move up as expected.

With resizeToAvoidBottomInset: false and an overlay editor on android:
Tapping the editor pops up the keyboard without any UI movement. When tapping outside to remove focus (to make the keyboard disappear), the overlay UI quickly moves up and then jumps back to its original position.

In summary: The position changes caused by the soft keyboard are completely unusable. I would prefer Flutter not adjust the position under any circumstances and let me control it manually. Unfortunately, Flutter always tries to do something stupid and damaging.

**Comment by JIanyag** [comment]
I have encountered the same problem. I don’t know whether flutter web can really be used in production environment or it is just a toy.
